src/hotspot/share/prims/whitebox.cpp

```diff
@@ -1962,6 +1962,14 @@ WB_ENTRY(jboolean, WB_IsSharingEnabled(JNIEnv* env, jobject wb))
   return UseSharedSpaces;
 WB_END
 
+WB_ENTRY(jint, WB_GetCDSGenericHeaderMinVersion(JNIEnv* env, jobject wb))
+  return (jint)CDS_GENERIC_HEADER_SUPPORTED_MIN_VERSION;
+WB_END
+
+WB_ENTRY(jint, WB_GetCDSCurrentVersion(JNIEnv* env, jobject wb))
+  return (jint)CURRENT_CDS_ARCHIVE_VERSION;
+WB_END
+
```

Comment:
Should these two functions be inside a `#if INCLUDE_CDS` block?

Reply:
We can return different values if INCLUDE_CDS is not defined. The functions should be available for java native functions, or a link error will be thrown when they are linked. Same answer to your next question.